x/valset-pref/msg_server_test.go

```diff
@@ -124,3 +124,82 @@ func (suite *KeeperTestSuite) TestSetValidatorSetPreference() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestDelegateToValidatorSet() {
+	suite.SetupTest()
+
+	// prepare validators to delegate to
+	preferences := suite.PrepareDelegateToValidatorSet()
+
+	tests := []struct {
+		name           string
+		delegator      sdk.AccAddress
+		coin           sdk.Coin
+		expectedShares []sdk.Dec
+		expectPass     bool
+		valSetExists   bool
+	}{
```

Comment:
would it be possible to add a test case that deals delegating coins that wouldn't be calculated perfectly under the ratio and test the rounding conditions? Ex) 739 or similar

Reply:
added it here [`45964da`](https://github.com/osmosis-labs/osmosis/pull/3260/commits/45964da70bb84d553d3cc72ce3a796411b58217f)

Reply:
cannot see it due to force push :( , can you point me to the specific test case?

Reply:
Yup, i changed the set-validator weights to have upto 3 decimal places https://github.com/osmosis-labs/osmosis/pull/3260/files#diff-102acff6e2fd0c8c3b2b95afe0c166f91100a1dcbc1204c40a0ff981823504a3R31